react/components/EstimateShipping.tsx

```diff
@@ -7,9 +7,13 @@ import { newAddress } from '../utils/address'
 
 const { AddressContainer, AddressRules, StyleguideInput } = components
 const { addValidation, removeValidation } = helpers
+interface InsertAddressResult {
+  success: boolean
+  orderForm: any
+}
 
 interface CustomProps {
-  insertAddress: (address: CheckoutAddress) => void
+  insertAddress: (coupon: CheckoutAddress) => Promise<InsertAddressResult>
```

Comment:
Shouldn't it be `address` instead of `coupon`?

Reply:
Done!